File: 998-insert-into-max-tree/solution.py
from typing import Optional
class TreeNode:
    def __init__(self, val, left, right) -> None:
        self.val = val
        self.left = left
        self.right = right    
class Solution:
    def insertIntoMaxTree(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        if root is None:
            return TreeNode(val, None, None)
        if val > root.val:
            return TreeNode(val, root, None)
        root.right = self.insertIntoMaxTree(root.right, val)
        return root

File: 998-insert-into-max-tree/solution.go
package main

import "gopractice/datastructure/Tree"

type TreeNode = Tree.TreeNode

func insertIntoMaxTree(root *TreeNode, val int) *TreeNode {
	if root == nil {
		return &TreeNode{val, nil, nil}
	}
	if val > root.Val {
		return &TreeNode{val, root, nil}
	} else {
		root.Right = insertIntoMaxTree(root.Right, val)
		return root
	}
}
